libsrc/apple2/cpuspeed.s

```diff
@@ -0,0 +1,40 @@
+;
+; Colin Leroy-Mira <colin@colino.net>, 2024
+;
+; void __fastcall__ apple2_slowdown(void)
+; void __fastcall__ apple2_speedup(void)
+;
+
+        .constructor    initcyareg, 8
+        .export         _apple2_slowdown
+        .export         _apple2_speedup
+        .import         apple2_ostype
+
+        .include        "apple2.inc"
+
+initcyareg:
+        lda     apple2_ostype
+        bpl     :+
+        lda     CYAREG
+        sta     cyareg_orig
+:       rts
```

Comment:
The user can change the system speed at any time using the Control Panel. It doesn't seem desirable to me to get the current speed at the start of the program.

Reply:
Oh, indeed. I thought it was at boot time only.

Reply:
But actual speed can certainly be changed at any time. But I seem to remember that I had problems with CYAREG not reflecting that actual speed...